src/parser/Expression.ts

```diff
@@ -41,11 +42,10 @@ export class BinaryExpression implements Expression {
         ];
     }
 
-    walk(visitor: ExpressionVisitor, parent?: Expression, cancel?: CancellationToken): void {
-        if (!cancel?.isCancellationRequested) {
-            visitor(this, parent);
-            this.left.walk(visitor, this, cancel);
-            this.right.walk(visitor, this, cancel);
+    walk(visitor: WalkVisitor, options?: WalkOptions) {
+        if (options?.walkExpressions) {
```

Comment:
I'm not sure we actually need this condition for these expressions. Generally the only reason you would call `expression.walk` is if you wanted to walk the expression. I put them there "just in case", but I could see an argument being made to exclude them to improve performance (ever so slightly).